examples/exex/rollup/src/db.rs

```diff
@@ -0,0 +1,231 @@
+            if wipe_storage {
+                tx.execute("DELETE FROM storage WHERE address = ?", (address.to_string(),))?;
+            } else {
+                for (key, value) in storage {
+                    tx.execute(
+                        "INSERT INTO storage (address, key, data) VALUES (?, ?, ?) ON CONFLICT(address, key) DO UPDATE SET data = excluded.data",
+                        (address.to_string(), key.to_string(), value.to_string()),
+                    )?;
+                }
+            }
+        }
+
+        for (hash, bytecode) in contracts {
+            tx.execute(
+                "INSERT INTO bytecode (hash, data) VALUES (?, ?) ON CONFLICT(hash) DO UPDATE SET data = excluded.data",
+                (hash.to_string(), bytecode.bytes().to_string()),
+            )?;
+        }
+
+        tx.commit()?;
+
+        Ok(())
+    }
+
+    pub fn increment_balance(&mut self, address: Address, value: U256) -> eyre::Result<()> {
+        let mut connection = self.connection();
+        let tx = connection.transaction()?;
+
+        let mut account = match tx.query_row::<String, _, _>(
+            "SELECT data FROM account WHERE address = ?",
+            (address.to_string(),),
+            |row| row.get(0),
+        ) {
+            Ok(account_info) => serde_json::from_str(&account_info)?,
+            Err(rusqlite::Error::QueryReturnedNoRows) => AccountInfo::default(),
+            Err(e) => return Err(e.into()),
+        };
+
+        account.balance += value;
+        account.nonce += 1;
+
+        tx.execute(
+            "INSERT INTO account (address, data) VALUES (?, ?) ON CONFLICT(address) DO UPDATE SET data = excluded.data",
+            (address.to_string(), serde_json::to_string(&account)?),
+        )?;
+        tx.commit()?;
+
+        Ok(())
+    }
+
+    pub fn decrement_balance(&mut self, address: Address, value: U256) -> eyre::Result<()> {
+        let mut connection = self.connection();
+        let tx = connection.transaction()?;
+
+        let account_info = tx.query_row::<String, _, _>(
+            "SELECT data FROM account WHERE address = ?",
+            (address.to_string(),),
+            |row| row.get(0),
+        )?;
+        let mut account: AccountInfo = serde_json::from_str(&account_info)?;
+
+        account.balance -= value;
+
+        tx.execute(
+            "INSERT INTO account (address, data) VALUES (?, ?) ON CONFLICT(address) DO UPDATE SET data = excluded.data",
+            (address.to_string(), serde_json::to_string(&account)?),
+        )?;
+        tx.commit()?;
+
+        Ok(())
+    }
+
+    pub fn get_account(&mut self, address: Address) -> eyre::Result<Account> {
+        let account_info = self.connection().query_row::<String, _, _>(
+            "SELECT data FROM account WHERE address = ?",
+            (address.to_string(),),
+            |row| row.get(0),
+        )?;
+        Ok(serde_json::from_str(&account_info)?)
+    }
```

Comment:
this uses `Account` but above we use `AccountInfo`, any reason why? in any case, i think you can use `get_account` in `decrement_balance`